### PR TITLE
Fix issues in hdd_reader

### PR DESCRIPTION
**Pull request by kmiya**
## What kinds of PR

- [X] Bug fix

## What is this PR

- Update the launch command
- Fix some issues detected by Coverity Scan
  - Resource leak (`new_sock`)
  - Uninitialized variables
  - Improper null termination (`buf`)
    - Ref. [CWE-170](https://cwe.mitre.org/data/definitions/170.html)
    - Almost all cases seem to be no problem, but just in case fixed it
- Fix some issues detected by Clang-Tidy
  - Deprecated headers
  - Use `using` instead of `typedef`
  - Redundant `void`
  - Use range-based `for`
  - Unused namespace alias
  - Use `nullptr` instead of `0` to initialize pointers

Do __not__ fix the other issues, for example:
- Use `std::array<>` instead of C-style array
- Do not use pointer arithmetic
- Do not use reinterpret_cast
- etc.

## How to check this PR
```shell
# build the package
# edit hdd.service
sudo systemctl restart hdd.service
# tty 1
ros2 launch system_monitor system_monitor.launch.xml
# tty 2
ros2 run rqt_runtime_monitor rqt_runtime_monitor
```
Seems to work correctly in my environment.
Screenshot from 2021-04-02 17-42-54
Screenshot from 2021-04-02 17-43-11

### CODE REVIEW

File: system/system_monitor/reader/hdd_reader/hdd_reader.cpp [hunk 1]
```diff
@@ -47,28 +47,26 @@
...
-namespace fs = boost::filesystem;
-
 // 7634-7647 Unassigned
 constexpr int PORT = 7635;
 
 /**
  * @brief HDD information
  */
-typedef struct
+using HDD_Info = struct
```
Comment (kenji-miyake) [inline system/system_monitor/reader/hdd_reader/hdd_reader.cpp:56]: or just `struct HDD_Info {};` ?

File: system/system_monitor/reader/hdd_reader/hdd_reader.cpp [hunk 2]
```diff
@@ -89,15 +87,15 @@ typedef struct
...
+};
 
 /**
  * @brief Attribute Table Format
  * @note For details please see the documents below.
  * - SMART Attribute Overview
  *   http://www.t13.org/Documents/UploadedDocuments/docs2005/e05171r0-ACS-SMARTAttributes_Overview.pdf
  */
-typedef struct __attribute__((packed))  // Minimize total struct memory 16 to 12
+using AttributeEntry = struct __attribute__((packed))  // Minimize total struct memory 16 to 12
```
Comment (kenji-miyake) [inline system/system_monitor/reader/hdd_reader/hdd_reader.cpp:98]: How about using `#pragma pack`? Is it different from `__attribute__((packed))`?

Comment (ito-san) [reply system/system_monitor/reader/hdd_reader/hdd_reader.cpp:98]: The usage of `#pragma pack(1)` is better style, I think.

Comment (kmiya) [reply system/system_monitor/reader/hdd_reader/hdd_reader.cpp:98]: The `__attribute__((packed))` might be a GCC-specific in ancient times.
Nowadays, we can use it in Clang as well as GCC. Seems to be no problem.
https://wandbox.org/permlink/p1fWzK5JqEVVDnaJ
Also, AUTOSAR C++14 says `#pragma` should not be used (Rule A16-7-1).

File: system/system_monitor/reader/hdd_reader/hdd_reader.cpp (hunk 1 above)
Comment (kmiya) [reply system/system_monitor/reader/hdd_reader/hdd_reader.cpp]: Oh, yeah! Fixed in 9a9612f83ad86dc2f070485d189f567a317b0f97.
`packed` still works.
https://wandbox.org/permlink/hZxGP08FUcCF1BLP

File: system/system_monitor/reader/hdd_reader/hdd_reader.cpp (hunk 2 above)
Comment (ito-san) [reply system/system_monitor/reader/hdd_reader/hdd_reader.cpp]: Understood. Looks like it should stay as it is! :+1:

**Review by kenji-miyake — APPROVED** [review]

**Pull request merged**